lib/models/obms.js

```diff
@@ -0,0 +1,183 @@
+// Copyrighr 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = ObmsModelFactory;
+
+ObmsModelFactory.$provide = 'Models.Obms';
+ObmsModelFactory.$inject = [
+    'Model',
+    'Services.Waterline',
+    '_',
+    'Services.Encryption',
+    'Constants',
+    'Promise',
+    'util',
+    'Errors',
+    'Assert'
+];
+
+function ObmsModelFactory (
+    Model,
+    waterline,
+    _,
+    encryption,
+    Constants,
+    Promise,
+    util,
+    Errors,
+    assert
+) {
+    var secretsPattern = _.reduce(Constants.Logging.Redactions, function(pattern, regex) {
```

Comment:
@yyscamper That's true, but it just reveals a problem with the way our models our structured.  In many places (including obms) we define model properties with `type: 'json'` and lose all power to place constraints and/or validations around that data at the model level.  In the case of obms, we have an external schema validator to help with this, but I think in most cases it indicates that we need to rethink the model design.  This problem is further complicated by a waterline limitation: https://github.com/balderdashy/sails-mongo/issues/44.